verifier/src/lib.rs

```diff
@@ -204,25 +222,37 @@ where
     // make sure that evaluations obtained by evaluating constraints over the out-of-domain frame
     // are consistent with the evaluations of composition polynomial columns sent by the prover
 
-    // read the out-of-domain evaluation frame sent by the prover and evaluate constraints over it;
-    // also, reseed the public coin with the OOD frame received from the prover
-    let ood_frame = channel.read_ood_evaluation_frame();
-    let ood_constraint_evaluation_1 = evaluate_constraints(&air, constraint_coeffs, &ood_frame, z);
-    public_coin.reseed(H::hash_elements(ood_frame.current()));
-    public_coin.reseed(H::hash_elements(ood_frame.next()));
+    // read the out-of-domain trace frames (the main trace frame and auxiliary trace frame, if
+    // provided) sent by the prover and evaluate constraints over them; also, reseed the public
+    // coin with the OOD frames received from the prover.
+    let (ood_main_trace_frame, ood_aux_trace_frame) = channel.read_ood_trace_frame();
+    let ood_constraint_evaluation_1 = evaluate_constraints(
+        &air,
+        constraint_coeffs,
+        &ood_main_trace_frame,
+        &ood_aux_trace_frame,
+        aux_trace_rand_elements,
+        z,
+    );
+    public_coin.reseed(H::hash_elements(ood_main_trace_frame.current()));
+    public_coin.reseed(H::hash_elements(ood_main_trace_frame.next()));
+    if let Some(ref aux_trace_frame) = ood_aux_trace_frame {
```

Comment:
OOD frames are handled differently on the prover and verifier side (main and aux segments are grouped together on the prover / split on the verifier side). This causes discrepancies when reseeding the public coin.